File: Algorithms/TwoCrystalBalls/TwoCrystalBalls.java
public class TwoCrystalBalls {
    static int solution(boolean[] arr) {
        int lenArr = arr.length;
        int step = (int) Math.floor(Math.sqrt(lenArr));
        int i = step;

        for (; i < lenArr; i += step) {
            if (arr[i])
                break;
        }

        i -= step;

        for (int j = 0; j <= step && i < lenArr; ++j, ++i) {
            if (arr[i])
                return i;
        }

        return -1;
    }

    public static void main(String[] args) {
        boolean[] arr = { false, false, false, false, false, true, true, true };
        System.out.println(solution(arr));
    }
}

File: Algorithms/TwoCrystalBalls/TwoCrystalBalls.py
import math

def two_crystal_balls(arr):
    len_arr = len(arr)
    jump = int(math.sqrt(len_arr))
    
    i = jump
    j = 0

    while i < len_arr:
        if arr[i]:
            break

        i += jump

    i -= jump

    while j <= jump and i < len_arr:
        if arr[i]:
            return i
        
        j += 1
        i += 1

    return -1
    
if __name__ == "__main__":
    arr = [False, False, False, False, True, True, True, True]
    print(two_crystal_balls(arr))